bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -1435,4 +1859,40 @@ static long fileName2LogId(String fileName) {
     static String logId2HexString(long logId) {
         return Long.toHexString(logId);
     }
+
+    /**
+     * Datastructure which maintains the status of logchannels. When a
+     * logChannel is created entry of < entryLogId, false > will be made to this
+     * sortedmap and when logChannel is rotated and flushed then the entry is
+     * updated to < entryLogId, true > and all the lowest entries with
+     * < entryLogId, true > status will be removed from the sortedmap. So that way
+     * we could get least unflushed LogId.
+     *
+     */
+    class RecentEntryLogsStatus {
+        private SortedMap<Long, Boolean> entryLogsStatusMap;
+        private long leastUnflushedLogId;
+
+        RecentEntryLogsStatus(long leastUnflushedLogId) {
+            entryLogsStatusMap = new TreeMap<Long, Boolean>();
+            this.leastUnflushedLogId = leastUnflushedLogId;
+        }
+
+        synchronized void createdEntryLog(Long entryLogId) {
+            entryLogsStatusMap.put(entryLogId, false);
+        }
+
+        synchronized void flushRotatedEntryLog(Long entryLogId) {
+            entryLogsStatusMap.replace(entryLogId, true);
+            while ((!entryLogsStatusMap.isEmpty()) && (entryLogsStatusMap.get(entryLogsStatusMap.firstKey()))) {
```

Comment:
```
map.entrySet().removeIf(entry -> {
     boolean flushed = entry.getValue();
     if (flushed) {
         leastUnflushedLogId = entry.getKey() + 1;
     }
     return flushed;
});
```

Reply:
here we are not removing all the entries whose value is true. We are removing just the entries which has true value untill the first false value. But yeah, probably we can use streams in some form, for simplicity sake lets keep as it is. May be I should add a comment describing what I'm intending to do here.